disperser/apiserver/disperse_blob_v2.go

```diff
@@ -178,6 +178,10 @@ func (s *DispersalServerV2) validateDispersalRequest(
 		return nil, errors.New("payment metadata is required")
 	}
 
+	if s.ReservedOnly && blobHeader.PaymentMetadata.CumulativePayment.Sign() != 0 {
```

Comment:
Do you need to check that the reservation_period (or timestamp lately) is not set?
Otherwise, it'll be treated as reserved, even though cumulative_payment is also set (reserved takes precedence).

Reply:
I don't think the check is needed, because I've been thinking adding timestamp to ondemand payments as well for smarter global ondemand ratelimiting, then the only clear separation between ondemand and reservation is `cumulative_payment`. This is also currently how meterer determines the type of requests

Reply:
It doesn't match what the API says: https://github.com/Layr-Labs/eigenda/blob/master/api/proto/common/v2/common_v2.proto#L66-L73

Reply:
ooof thanks for pointing it out. The API docs need updating

Reply:
Can you update it to make them consistent?

Reply:
yeah, I overhauled the section this afternoon, pls lmk if it looks better https://github.com/Layr-Labs/eigenda/pull/1570